### PR TITLE
[DEV-5735] metrics jira controller multiple groups

### PR DESCRIPTION
**Pull request by gaetano-guerriero**
This change is [Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3371)

### CODE REVIEW

File: server/athenian/api/controllers/jira_controller.py [hunk 1]
```diff
@@ -122,15 +121,10 @@ async def filter_jira_stuff(request: AthenianWebRequest, body: dict) -> web.Resp
...
-        request.sdb,
-        request.mdb,
-        request.pdb,
-        request.cache,
     )
     if filt.projects is not None:
-        projects = await resolve_projects(filt.projects, jira_ids.acc_id, request.mdb)
-        projects = {k: projects[k] for k in jira_ids.projects if k in projects}
-        jira_ids = JIRAConfig(jira_ids.acc_id, projects, jira_ids.epics)
+        projects = jira_conf.project_ids_map(filt.projects)
```
Comment (gaetano-guerriero) [inline server/athenian/api/controllers/jira_controller.py:126]: no need to touch db again JIRAConfig has all the projects, we can compute the subset

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]
> Reviewed 1 of 10 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3371)* status: 1 of 10 files reviewed, 2 unresolved discussions (waiting on @gaetano-guerriero)

___
*[`server/athenian/api/controllers/jira_controller.py` line 1438 at r1](https://reviewable.io/reviews/athenianco/athenian-api/3371#-NOOVX0a88PVdYiDp2FY:-NOOVX0a88PVdYiDp2FZ:bny9mvq) ([raw file](https://github.com/athenianco/athenian-api/blob/195703fb045631dc9029412a08d6db19fd852fa2/server/athenian/api/controllers/jira_controller.py#L1438)):*
> ```Python
>         res_groups = [None] * len(split_labels)
> 
>     granularities = metrics_req.granularities or ["all"]
> ```

`granularities` is a required field, there must be at least one element. We shouldn't do `or ["all"]`

**Review by gaetano-guerriero — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3371)* status: 1 of 12 files reviewed, 3 unresolved discussions (waiting on @vmarkovtsev)

___
*[`server/athenian/api/controllers/jira_controller.py` line 1438 at r1](https://reviewable.io/reviews/athenianco/athenian-api/3371#-NOOVX0a88PVdYiDp2FY:-NOQEjvy6p0CUj72BNTF:bxkqlmj) ([raw file](https://github.com/athenianco/athenian-api/blob/195703fb045631dc9029412a08d6db19fd852fa2/server/athenian/api/controllers/jira_controller.py#L1438)):*
<details><summary><i>Previously, vmarkovtsev (Vadim Markovtsev) wrote…</i></summary><blockquote>

`granularities` is a required field, there must be at least one element. We shouldn't do `or ["all"]`
</blockquote></details>

Done.
I'll avoid this https://athenianco.atlassian.net/browse/DEV-5981 by fixing the spec then with `minItems: 1`
___
*[`server/athenian/api/internal/jira.py` line 107 at r1](https://reviewable.io/reviews/athenianco/athenian-api/3371#-NOOWgS--69-DOV2dC0G:-NOQEi0U7qS15Xodq71c:b-896fix) ([raw file](https://github.com/athenianco/athenian-api/blob/195703fb045631dc9029412a08d6db19fd852fa2/server/athenian/api/internal/jira.py#L107)):*
<details><summary><i>Previously, vmarkovtsev (Vadim Markovtsev) wrote…</i></summary><blockquote>

I checked: we convert this to a frozenset everywhere we use it. Let's return a frozenset directly then.
</blockquote></details>

Done.

**Review by vmarkovtsev — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3371)* status: 1 of 12 files reviewed, all discussions resolved

**Pull request merged**